lib/puppet/configurer.rb

```diff
@@ -229,6 +229,9 @@ def run_internal(options)
       end
 
       execute_prerun_command or return nil
+
+      options[:report].code_id = catalog.code_id
+      options[:report].catalog_uuid = catalog.catalog_uuid
```

Comment:
Also I had some inner doubts about whether this would break anything or not when running with different agent / server version combos, as old catalogs wouldn't have these elements. I landed on thinking that it *wouldn't*, since I believe we are only preserving old agents talking to newer masters.